wgsl/index.bs

```diff
@@ -1144,6 +1145,8 @@ unary_expression
       OpSNegate
       OpFNegate
   | BANG unary_expression
+      OpLogicalNot
+  | TILDE unary_expression
```

Comment:
Did you consider using the same `BANG` operator for this? Since one applies to logical expressions, and the other - for numeric ones, and there is no ambiguity, it's not obvious to me that this requires a new operator.

Reply:
This behavior can be surprising when moving from most languages to Rust. While it is less redundant, and is theoretically sound, I will say I've also found it surprising there.

Reply:
I was just trying to keep with C tradition.
I believe in C there would be ambiguity because it has implicit conversion-to-bool via test-against-zero